govcd/system.go

```diff
@@ -502,6 +505,94 @@ func QueryProviderVdcByName(vcdCli *VCDClient, name string) ([]*types.QueryResul
 	return results.Results.VMWProviderVdcRecord, nil
 }
 
+// QueryNetworkPoolByName finds a network pool by name
+func (vcdClient *VCDClient) QueryNetworkPoolByName(name string) (*types.QueryResultNetworkPoolRecordType, error) {
+	results, err := vcdClient.QueryWithNotEncodedParams(nil, map[string]string{
+		"type":   "networkPool",
+		"filter": fmt.Sprintf("(name==%s)", name),
+	})
+	if err != nil {
+		return nil, err
+	}
+	for _, networkPool := range results.Results.NetworkPoolRecord {
```

Comment:
I am not definitely sure is right approach. Cause `name==%s` already do this. Though name may be regexp when result will be quite different. And if resource can have a few instances with same name? Same comment is valid for other changed Query functions!

Reply:
Yeah. I also started to think what can happen in case when multiple objects are allowed to have the same name. (IIRC so far we know independent disks can have the same names)

Reply:
I know about independent disks, but in this case we know that the objects we are looking for have unique names.

I don't particularly care about having a Query*ByName with a single result at the moment, since these functions are unused. I proposed this rewrite because the original one looks wrong to me, because the name is deceiving and because it returns a `nil` error when not found.
 
My goal with this PR is to have a list of Provider VDCs, Network Pools, and Storage Profiles, which are needed for automation, and we lack the ability of fetching the list.

However, these functions will become important when (or if) we implement data sources for such entities, or when we implement data sources for other entities that need to be retrieved through Query functions. At that stage, we need a way of getting those entities by ID, and sometimes by name.

If there is no agreement on this function, I'll remove it and only keep the list functions only.

Reply:
For what purpose these functions could be used it's another story. One e.g. can be that returning attributes are different. If you want to now which ext network egtw is using - you have to use query result, cause direct GET won't provide such information. Going back regarding implementation, IMHO, isn't right if we query by name and expect one result when check len() and return result. Currectly it is doing same check against name and qeury return a few results it's choosing only first result. And such behaviour I don't like

Reply:
I have removed the methods `vcdClient.QueryNetworkPoolByName`, `vcdClient.QueryProviderVdcStorageProfileByName`, and `vcdClient.QueryProviderVdcByName`, and the deprecation notice from the corresponding functions.